x-pack/plugins/ml/public/application/components/data_recognizer/data_recognizer.js

```diff
@@ -29,6 +29,11 @@ export class DataRecognizer extends Component {
     // once the mount is complete, call the recognize endpoint to see if the index format is known to us,
     ml.recognizeIndex({ indexPatternTitle: this.indexPattern.title })
       .then((resp) => {
+        // Sort results by title prior to display
+        resp.sort((res1, res2) => {
+          return res1.title.localeCompare(res2.title);
+        });
```

Comment:
shall this sorting happens server-side? in case of any solution plugin uses this endpoint we can guarantee the order.

Reply:
Good point. I have added a sort by `id` to the server-side `/api/ml/modules/recognize/{indexPatternTitle}` endpoint  in https://github.com/elastic/kibana/pull/86250/commits/84dbb53cfda8f16a017178fc379c3a3ac09e955e.

Sorting by `title` looks better IMO in the ML job wizard / data visualizer sidebar, but sorting by `id` seems more logical for the `recognize` endpoint, so I have left the client side sort in too.